src/rpc/blockchain.h

```diff
@@ -5,9 +5,13 @@
 #ifndef BITCOIN_RPC_BLOCKCHAIN_H
 #define BITCOIN_RPC_BLOCKCHAIN_H
 
-#include <vector>
-#include <stdint.h>
 #include <amount.h>
+#include <sync.h>
+
+#include <stdint.h>
+#include <vector>
+
+extern RecursiveMutex cs_main;
```

Comment:
@MarcoFalke Do you use `RecursiveMutex` instead of `CCriticalSection` on purpose?